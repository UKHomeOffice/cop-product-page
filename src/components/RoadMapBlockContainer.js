import React from "react";
import RoadMapBlock from "./roadmap-block";

const RoadMapBlockContainer = (props) => (
  <div>
    {props.content.map((value, index) => {
      return <RoadMapBlock key={index} text={value} bannerColour={props.bannerColour} />
    })}
  </div>
);

export default RoadMapBlockContainer;