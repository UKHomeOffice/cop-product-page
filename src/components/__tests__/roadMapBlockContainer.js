import React from "react";
import renderer from "react-test-renderer";
import RoadMapBlockContainer from "../RoadMapBlockContainer";


describe("roadmapBlockContainer", () => {

  const roadMapItem = {
    streamLine: {
      measure: ["Time saved", "Daily and monthly users", "% of users on mobile", "# of forms completed"],
      now: ["Intel Referral", "Events at the border", "Intel workflow", "sGMR", "Firmstep Migration*"],
      next: ["NCC Task Management", "Capacity Planning (OAR)", "RefDataV2", "Integrity Checks", "Cerberus Task Management*"],
      later: ["Record Refusal for entry (IS81)", "OAR Enhance", "Man DecV2", "Ops Planning Tools"]
    },
    inform: {
      measure: ["# reports available", "# analysts using data", "# of cases to stakeholders", "faster availability of data"],
      now: ["BI Integration", "Analyst Data (self serve)", "PRAU feed V2", "Product Page"],
      next: ["Shift Debrief", "Sit Rep View", "Mandatory Declarations Viewer", "Embedded BI dashboards in COP"],
      later: ["Service Perf Dashboard", "Ops Dashboard", "Intel Dashboard"]
    },
    trusted: {
      measure: ["Availability", "# grey IT systems removed", "Cost per transaction", "User satisfaction"],
      now: ["Image Capture (Smartphone)", "Workflow collaboration Tool (Cawemo)", "24x7 Support", "Service Readiness Review", "Secure, Run, Continuous Improve"],
      next: ["eDcoument read & verify", "Workflow Engine Upgrade 7.13", "Optimise Tool", "User Documentation V2", "Secure, Run, Continuous Improve"],
      later: ["Fingerprint capture", "Workflow Engine Upgrade 7.14", "Release Comms", "Secure, Run, Continuous Improve"]
    }
  };

  it("renders correctly", () => {
    const tree = renderer
      .create(<RoadMapBlockContainer content={roadMapItem.streamLine.now} bannerColour={'#28afb0'} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Returns the correct RoadMapBlockContainer 'content' and bannerColour prop passed in", () => {
    const tree = renderer
      .create(<RoadMapBlockContainer content={roadMapItem.streamLine.now} bannerColour={'#28afb0'}/>);
    const testInstance = tree.root;
    expect(testInstance.props).toHaveProperty('content', ["Intel Referral", "Events at the border", "Intel workflow", "sGMR", "Firmstep Migration*"]);
    expect(testInstance.props).toHaveProperty('bannerColour', '#28afb0');
  });
});