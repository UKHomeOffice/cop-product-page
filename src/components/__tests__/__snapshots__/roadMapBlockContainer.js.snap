// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`roadmapBlockContainer renders correctly 1`] = `
<div>
  <div
    className="rectangle rectangle-block"
    style={
      Object {
        "borderLeft": "solid 6px#28afb0",
      }
    }
  >
    <div
      className="roadmap-block-text"
    >
      Intel Referral
    </div>
  </div>
  <div
    className="rectangle rectangle-block"
    style={
      Object {
        "borderLeft": "solid 6px#28afb0",
      }
    }
  >
    <div
      className="roadmap-block-text"
    >
      Events at the border
    </div>
  </div>
  <div
    className="rectangle rectangle-block"
    style={
      Object {
        "borderLeft": "solid 6px#28afb0",
      }
    }
  >
    <div
      className="roadmap-block-text"
    >
      Intel workflow
    </div>
  </div>
  <div
    className="rectangle rectangle-block"
    style={
      Object {
        "borderLeft": "solid 6px#28afb0",
      }
    }
  >
    <div
      className="roadmap-block-text"
    >
      sGMR
    </div>
  </div>
  <div
    className="rectangle rectangle-block"
    style={
      Object {
        "borderLeft": "solid 6px#28afb0",
      }
    }
  >
    <div
      className="roadmap-block-text"
    >
      Firmstep Migration*
    </div>
  </div>
</div>
`;
